.github/workflows/build-deploy-doc.yml

```diff
@@ -1,5 +1,11 @@
 name: Build and Deploy documentation site
 
+# This workflow is a customized workflow to deploy docs site to Cloudflare
+# The reason we can no longer use Cloudflare's git integration to deploy docs
+# site anymore is that Cloudflare's git integration would only check out
+# documentation repo, whereas we need to check out both `endojs/endo` and
+# `agoric/agoric-sdk` repo to build the docs site.
```

Comment:
I wonder about using submodules.

It would be nice to be able to get snippets from dapp-agoric-basics, for example.

Reply:
Yeah I considered it but in my past experience I've only been frustrated about needing to update the hashes for submodules and make git commits. That said, there's not really a good solution whenever we are "orchestrating" multiple repos.

I'm happy to give it another try if there's some submodule-fu I'm missing though.

Reply:
Maybe the submodule friction is a good thing?

Looks like this workflow is checking out HEAD of the other two repos, but that's not what people can use. I propose that the docs reflect what's in Mainnet. To do that requires a PR on this repo to specify the the branch or tag in the other repos. Expressing that as a git submodule makes sense and it sounds like it would obviate the need to define this additional workflow.

Reply:
> I propose that the docs reflect what's in Mainnet.

There's some appeal to that, but then how to we make progress on generated reference docs?

Reply:
The reference docs from each repo can publish anywhere. But this PR is about docs.agoric.com, right? Shouldn't that URL only show you what's in Mainnet?

Reply:
again, yes, there's some appeal to that, but running typedoc on the mainnet version of our code yields mostly garbage. So should the generated reference docs on dogs.agoric.com be garbage until our fixes to the jsdoc in agoric-sdk make it to mainnet?

Reply:
Running typedoc on the mainnet version of the code won't work at all, in fact, will it?

Reply:
>So should the generated reference docs on dogs.agoric.com be garbage until our fixes to the jsdoc in agoric-sdk make it to mainnet?

I'm not proposing that. I'm speaking to whether this repo sources from HEAD or a git submodule. My claim is that sourcing HEAD of agoric-sdk is untenable and so it shouldn't be how this PR is implemented. IIUC, there's no need for a new workflow if the reop used git submodules. Given that we need some way to specify a tag of the source repos, I propose that be how this is implemented.

What tag/version of the other repos to source, I think can change over time. I agree that u13 would be unworkable. Probably same for u14. But maybe u15 can get the docs improvements that have been going into master. Again, independent of my request that the repos are sourced by a tag instead of HEAD.

Reply:
> I agree that u13 would be unworkable. Probably same for u14. But maybe u15 can get the docs improvements that have been going into master.

ok. yes, that's pretty much what I'm thinking.

Reply:
This discussion is helpful, I'll look into submodules